be/src/runtime/exec_env.h

```diff
@@ -219,6 +225,9 @@ class ExecEnv {
     HeartbeatFlags* _heartbeat_flags = nullptr;
 
     PluginMgr* _plugin_mgr = nullptr;
+
+    std::map<std::string, FrontendStartInfo*> _frontends_start_time;
```

Comment:
use unordered map.
And this structure need to be protected by lock, you can try using `phmap::parallel_flat_hash_map`, which is a concurrent hash map of C++. You can refer to `be/src/util/brpc_stub_cache.h`

Reply:
Thanks for your advice,
But `_frontends_start_time` I understand that there is no need to lock, it will only be modified when the BE is processing the heartbeat, and a BE process has only one thread to process the heartbeat.

Reply:
done, Thats right